quantized embedding: make error message clearer

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66051
* #66050
* #66049

Summary:

Make the error message clearer when quantized embedding is converted
with an unsupported dtype. This is helpful when debugging quantization
errors on new models.

Test Plan:

```
class M(nn.Module):
    def __init__(self):
        super().__init__()
        self.embedding = nn.Embedding(1, 1)

m = M().eval()
m.qconfig = torch.quantization.QConfig(
    activation=torch.quantization.MinMaxObserver.with_args(dtype=torch.qint8),
    weight=torch.quantization.MinMaxObserver.with_args(dtype=torch.qint8))
m.embedding.qconfig = m.qconfig
mp = torch.quantization.prepare(m)
mq = torch.quantization.convert(m)
// error message now includes the incorrect dtype
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31472848](https://our.internmc.facebook.com/intern/diff/D31472848)

vkuzo committed:
quantized embedding: make error message clearer

Summary:

Make the error message clearer when quantized embedding is converted
with an unsupported dtype. This is helpful when debugging quantization
errors on new models.

Test Plan:

```
class M(nn.Module):
    def __init__(self):
        super().__init__()
        self.embedding = nn.Embedding(1, 1)

m = M().eval()
m.qconfig = torch.quantization.QConfig(
    activation=torch.quantization.MinMaxObserver.with_args(dtype=torch.qint8),
    weight=torch.quantization.MinMaxObserver.with_args(dtype=torch.qint8))
m.embedding.qconfig = m.qconfig
mp = torch.quantization.prepare(m)
mq = torch.quantization.convert(m)
// error message now includes the incorrect dtype
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on torch/nn/quantized/modules/embedding_ops.py:
```diff
@@ -149,7 +149,8 @@ def from_float(cls, mod):
 
-        assert dtype == torch.quint8, 'The only supported dtype for nnq.Embedding is torch.quint8'
+        assert dtype == torch.quint8, \
+            f'The only supported weight dtype for nnq.Embedding is torch.quint8, got {dtype}'
```
Can you also make this change for nnq.EmbeddingBag in the same file?

vkuzo committed:
Update on "quantized embedding: make error message clearer"


Summary:

Make the error message clearer when quantized embedding is converted
with an unsupported dtype. This is helpful when debugging quantization
errors on new models.

Test Plan:

```
class M(nn.Module):
    def __init__(self):
        super().__init__()
        self.embedding = nn.Embedding(1, 1)

m = M().eval()
m.qconfig = torch.quantization.QConfig(
    activation=torch.quantization.MinMaxObserver.with_args(dtype=torch.qint8),
    weight=torch.quantization.MinMaxObserver.with_args(dtype=torch.qint8))
m.embedding.qconfig = m.qconfig
mp = torch.quantization.prepare(m)
mq = torch.quantization.convert(m)
// error message now includes the incorrect dtype
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
